lerobot/scripts/record_and_replay_teleop_dataset.py

```diff
@@ -0,0 +1,769 @@
+            robot.connect()
+
+        while True:
+            observation = robot.capture_observation()
+            positions = observation["joints"]
+            assert len(joint_commands) == len(positions)
+            for i in range(len(joint_commands)):
+                joint_commands[i] = positions[i]
+
+    joint_commands = Array('d', [0, 0, 0, 0, 0])
+    run_robot_process = Process(target=update_controls, args=(joint_commands))
+
+    # Separate thread to copy arm data to teleop_command (this shares compute time with simulation).
+    from threading import Thread
+
+    def copy_data(joint_commands: Array, teleoperation_action, is_stopping: bool):
+        while not is_stopping:
+            if teleoperation_action is None:
+                time.sleep(0.01)
+                continue
+
+            assert len(joint_commands) == len(teleoperation_action)
+            for i in range(len(joint_commands)):
+                teleoperation_action[i] = joint_commands[i]
+
+    copy_data_thread = Thread(target=copy_data, args=(
+        joint_commands, episode_state.teleoperation_action, episode_state.is_stopping))
+
+    def start_fn():
+        listener.start()
+        run_robot_process.start()
+        copy_data_thread.start()
+
+    def stop_fn():
+        listener.stop()
+        run_robot_process.join()
+        copy_data_thread.join()
+        del robot
+
+    return [start_fn, stop_fn]
+
+
+def set_up_teleop(teleop_method: str):
+    start_fn = None
+    stop_fn = None
+
+    if teleop_method == "keyboard":
+        (start_fn, stop_fn) = set_up_keyboard_teleop()
+    elif teleop_method == "arm":
+        (start_fn, stop_fn) = set_up_arm_teleop()
+    else:
+        raise Exception(
+            "A teleoperation method must be selected (Currently only 'keyboard' teleop is suppored)!")
+
+    if not isinstance(start_fn, types.FunctionType):
+        raise ValueError(
+            "start_fn is not callable but a teleoperation system was initialized.")
+
+    if not isinstance(stop_fn, types.FunctionType):
+        raise ValueError(
+            "stop_fn is not callable but a teleoperation system was initialized.")
+
+    return [start_fn, stop_fn]
+
+
+##########################################################################################
+# Running the task
+class TaskParameters(NamedTuple):
+    cube_file_path: str | None = None
+
+
+def construct_and_set_up_env(*, task_parameters: TaskParameters):
+    lerobot.common.utils.utils.set_global_seed(args.random_seed)
+    # Create the gym environment - check the kwargs in gym_real_world/gym_environment.py
+    env = gym.make(args.env_name,
+                   disable_env_checker=True,
+                   observation_mode="both",
+                   action_mode="ee" if args.teleop_method == "keyboard" else "joint",
+                   render_mode="human",
+                   cube_file_path=task_parameters.cube_file_path
+                   )
+
+    return env
```

Comment:
Regarding these gym arguments. I think if we want to make this script support different simulators then we should find a more general format. For example, in Mujoco the cube is usually defined the xml file that describes the scene. so we could get the gym arguments from the config files and do:
```
env = gym.make(args.env_name, **gym_kwargs)
```

Reply:
Great Idea, let me know where is best to put those configs (in `lerobot/configs/env/` now).